### PR TITLE
Transition from DTD/EBNF to RELAX NG Compact schema

### LINKED ISSUE #11: Transition to RELAX NG Compact schema language
Since the grammar inside of the standard is described with the EBNF,
this transition would only require swapping the DTD with RNC in Appendix B.
Additionally, it would need appropriate wording in introduction to justify the EBNF.

### PR DESCRIPTION
**Pull request by rakhimov**
As presented at the workshop,
RelaxNG Compact (RNC) is a modern and convenient schema language
capable of fully describing features of the MEF
unlike the DTD.

The RNC syntax is very much the BNF-like readable,
leveraging regular expression operators.
The transition from the DTD to the RNC
should ease introduction of new MEF features.

The MEF schema in the RNC language is already provided
thanks to the early effort in the SCRAM project.
Only work necessary was to swap the schema in the Appendix B,
and update the wording in Chapter 2: Introduction.

~~The only current minor-drawback
is that syntax highlighter Pygments
doesn't yet provide support for RNC,
so it looks like a plain text or the EBNF.~~
*(Available with Pygments 2.2)*

Closes #11

### CODE REVIEW

**Comment by rakhimov** [comment]
Syntax highlighting will be provided in upcoming Pygments 2.2

**Comment by rakhimov** [comment]
A good tutorial for reviewers of this PR: http://relaxng.org/compact-tutorial-20030326.html
This tutorial is also referenced in the MEF instead of giving its own custom one.

**Comment by rakhimov** [comment]
The work on conversion to RELAX NG Compact (RNC) is complete.
The EBNF and DTD are removed in favor of the RNC.
As a result, the grammar-to-schema synchronization burden is eliminated altogether.
Pesky errors in the ENBF are gone for good.
The redundant appendix of the full MEF EBNF is removed.

Thanks to excellent tutorials of the RNC and its self-descriptiveness,
the MEF specification is free from the burden of inventing its own tutorials.
Thus, the tutorial appendix is gone.

Following the Single Point of Truth (SPOT) principle,
all the grammar specifications inside of the MEF
are automatically compiled into the final RNC XML schema of the MEF.
It is always up-to-date.

Moreover, there's a requirement for all the schemas in the specification
to be correct and complete.
In other words, no more sketches or partial grammars.
Each construct requires its own full grammar to be presented with its section/description.

To bomb the last Appendix with the full schema of the MEF,
the schema (or rather schemas in various formats)
is provided in the special repository referenced by the standard.
Furthermore, the reference schema in the RNC format
is provided together with the standard in the zip file.

This work removes a lot of accidental complexity out of the standardization process.
The goal is for the community to have easier (less error-prone) time working on the MEF.

This PR is ready to be merged.

**Comment by rakhimov** [comment]
The resulting html/pdf can be seen at https://rakhimov.github.io/mef-rnc/

**Review by mhibti — APPROVED** [review]
That is Ok for me. Thanks

**Comment by rakhimov** [comment]
Thanks @mhibti !
